### PR TITLE
Fix width computation in Loop::rewrite

### PR DESCRIPTION
**Pull request by sanxiyn**
Found while working on #1045.

### CODE REVIEW

File: src/expr.rs [hunk 1]
```diff
@@ -674,9 +674,13 @@ impl<'a> Rewrite for Loop<'a> {
             ControlBraceStyle::AlwaysNextLine => alt_block_sep.as_str(),
             ControlBraceStyle::AlwaysSameLine => " ",
         };
+        let block_width = try_opt!(width.checked_sub(label_string.len() + self.keyword.len() +
+                                                     extra_offset(&pat_expr_string, inner_offset) +
+                                                     1));
```
Comment (nrc) [inline src/expr.rs]: This seems right for the empty loop case, but wrong when we have statements in the loop. In that case, the width for the statement should be just the width minus the next block offset (which block.rewrite will take care of). Or am I misunderstanding what is going on here?

Comment (sanxiyn) [inline src/expr.rs]: Yes you are. Width passed to Block::rewrite is only used for the empty loop case (and for single line unsafe block, which doesn't apply here). In the non-empty case, width is reset to max width minus offset.

Comment (nrc) [inline src/expr.rs]: OK, sweet, thanks for the explanation. Could you add a comment here summarising that please? And also on the Rewrite impl for Block.

**Comment by sanxiyn** [comment]
Added comments.

**Comment by nrc** [comment]
Thanks for the PR!

**Pull request merged**